Feat: 커피챗 신청 CRUD 기능 구현

jooyeon0907 opened the pull request:
### 변경사항
<!-- 이 PR에서 어떤 점들이 변경되었는지 기술해주세요. 가급적이면 AS-IS, TO-BE를 활용해서 작성해주세요. -->

#### AS-IS
- 커피챗 신청 기능이 구현되지 않음

#### TO-BE
 - **커피챗 신청 기능 추가**  
    - 사용자가 원하는 커피챗을 신청할 수 있는 API 추가  
    - 신청 시 진행 희망 시간과 질문 사항을 포함한 신청 내용 작성하여 요청
    - **포인트 차감 정책 적용 논의 필요** 

 - **커피챗 신청 목록 조회 기능**  
     - 사용자가 신청한 커피챗 목록을 조회할 수 있는 API 추가  

-  **커피챗 신청 수정 기능**  
    - 사용자가 신청한 커피챗 정보를 수정할 수 있는 API 추가  
    - **멘토가 아직 수락/거절하지 않은 대기 상태의 커피챗만 수정 가능**  
    - **신청 내용(`content`)만 수정 가능, 커피챗 시간(`coffeeChatStartTime`, `coffeeChatEndTime`)은 수정 불가**  

-  **커피챗 신청 취소(삭제) 기능**  
    - 사용자가 신청한 커피챗을 취소(삭제)할 수 있는 API 추가  

-  **유효성 검사 추가**  
    - `requestDTO`에 `@NotNull` 어노테이션 적용  

-  **추가된 엔티티 및 서비스**  
    - 커피챗 신청 관련 **컨트롤러, 서비스, DTO 클래스 추가**  

-  **테스트 데이터 설정**  
    - 테스트용 데이터에서 `userId`를 `1`로 설정하여 검증 진행  
    - 추후 **사용자 인증 적용 시, 인증된 사용자 정보에서 `userId`를 추출하도록 변경 예정**  

-  **네이밍 규칙 정리**  
    - **`Application` → `App`으로 축약하여 가독성 개선**  
    - ex) `coffeeChatApplicationId` → `coffeeChatAppId`  

- **필드명 변경**  
  - `requestStartTime`, `requestEndTime` → `coffeeChatStartTime`, `coffeeChatEndTime` (의미 명확화)  

---


### 추가 논의 사항
- **커피챗 신청 등록 시 포인트 차감 정책 의견**  
    - 현재는 신청 등록 시 포인트 차감 없이 진행하는데,
    - 포인트 회수 및 차감 정책에 대한 추가 논의가 필요해 보입니다.
    - **무분별한 신청을 방지하기 위해 신청 성공 시 포인트를 차감하면 어떨까 싶습니다.**  
        - 신청 등록 시 포인트 차감 → 멘토가 거절하면 포인트 회수  
        - 멘토가 승인하면 포인트 차감 유지  

 - **패키지 구조 정리**  
    - 현재 `coffeeChat` 패키지 내 구현하였는데,
    - **`coffeeChatApplication` 패키지로 분리하는 것이 더 적절하면 반영하여 수정 예정입니다.**   
    - @geonuu09 님 이 부분 확인 부탁드려요

---

### 테스트
<!-- 본 변경사항이 테스트되었는지 기술해주세요. -->
- [ ] **단위 테스트 진행**  
- [x] **API 테스트 진행**  
  - **커피챗 신청 등록**
    - 등록 성공 여부 확인  
    - ![등록성공](https://github.com/user-attachments/assets/d83e113e-aaac-4969-8835-a2413bfca732)

    - 예외 처리  
      - 신청하려는 커피챗 정보가 존재하지 않음 → `"커피챗을 찾을 수 없습니다."`  
      - ![커피챗없음](https://github.com/user-attachments/assets/304dbf13-e8d9-4c82-98e4-9bda1fa5867f)


  - **나의 커피챗 신청 목록 조회**  
    - 목록 조회 성공 여부 확인  
    - ![목록](https://github.com/user-attachments/assets/40350987-77d3-4d69-9643-d984fe72b012)

  - **커피챗 신청 수정**  
    - 수정 성공 여부 확인  
    - ![수정](https://github.com/user-attachments/assets/5e47f06b-f88f-42a0-9d43-85032ba36417)

    - 예외 처리  
      - 커피챗 신청이 존재하지 않음 → `"커피챗 신청 내역을 찾을 수 없습니다."`  
      - ![신청내역없음](https://github.com/user-attachments/assets/5bfbf5bc-244a-4884-825b-72016292c0cd)

      - 수정 요청한 사용자와 신청자가 다름 → `"사용자가 작성한 커피챗 신청 내역이 아닙니다."`  
      - ![사용자신청자아님](https://github.com/user-attachments/assets/6ca71ed0-c565-43f4-a6a4-1b6f92a3668f)

      - 수정하려는 신청 내역에 해당하는 커피챗 정보가 존재하지 않음 → `"커피챗을 찾을 수 없습니다."`  

  - **커피챗 신청 삭제**  
    - 삭제 성공 여부 확인  
    - ![image](https://github.com/user-attachments/assets/db97cbc6-9dd6-4e76-9cd1-2e122c836c51)

    - 예외 처리  
      - 커피챗 신청이 존재하지 않음 → `"커피챗 신청 내역을 찾을 수 없습니다."`  
      - 삭제 요청한 사용자와 신청자가 다름 → `"사용자가 작성한 커피챗 신청 내역이 아닙니다."`  
      - 삭제하려는 신청 내역에 해당하는 커피챗 정보가 존재하지 않음 → `"커피챗을 찾을 수 없습니다."`

jooyeon0907 committed:
feat: 커피챗 신청 CRUD 기능 구현
- 커피챗 신청 CRUD 기능 구현
- coffeeChatApplicationId 가 길어 coffeeChatAppId 로 축약하여 사용

jooyeon0907 committed:
Merge branch 'develop' of github.com:BootTalk-Service/boot-talk-back into feat/coffeechat_application

jooyeon0907 committed:
chore: TODO 제거

shCHO9801 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/dto/CoffeeChatApplicationResponseDto.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    // 목록 조회에서는 content를 포함하지 않음 (불필요한 데이터 전송 방지)
+    public static CoffeeChatApplicationResponseDto fromListAll(CoffeeChatApplication coffeeChatApp) {
```
이런식으로 데이터를 관리할수도 있겠군요!
새로운 아이디어 배워갑니다

shCHO9801 reviewed (commented):
고생하셨습니다!

jooyeon0907 committed:
fix: CoffeeChatApplicationResponseDto 수정
- status 타입 String -> StatusType 수정
- user 이름 필드가 name -> userName 으로 변경된 사항에 맞게 수정

jooyeon0907 committed:
fix: 커피챗 신청 상세 보기 API 삭제
- 커피챗 신청 목록 조회 API 응답 값만으로 상세 정보 표시 가능하여 별도의 상세 보기 API 제거
- 프론트엔드에서 모달을 통해 상세 내용을 보여주므로, API 호출 없이 기존 데이터 활용 가능

geonuu09 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/dto/CoffeeChatApplicationResponseDto.java:
```diff
@@ -0,0 +1,45 @@
+        );
+    }
+
+    // 목록 조회에서는 content를 포함하지 않음 (불필요한 데이터 전송 방지)
```
주연님, 그럼 제가 신청할 때 작성한 content는 수정하는 경우에만 확인할 수 있겠네요? 
신청 당시에 작성 한 내용을 확인만 하고싶은 경우도 있을 것 같아 코멘트 남깁니다~

jooyeon0907 replied:
신청 목록에서 신청 내역 클릭 시, 모달 화면 띄워서 상세 내용 표시하기로 했었어요 (회의록에 기록되어있습니다!)
figma를 보면 커피챗 보낸 신청 목록에 신청 내용(content) 이 표시가 안되어있어서 해당 데이터가 사용되지 않은 줄 알고 null 로 표시하기 위해 분리를 한 거였는데, 프론트 담당자분과 얘기해보니 상세 API 호출 없이 목록 조회 API 요청을 통해 받은 데이터로 활용 가능하다고 하셔서 이 부분 수정하여 반영하였습니다 참고부탁드려요! 
코멘트 감사합니다~

geonuu09 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/dto/CoffeeChatApplicationResponseDto.java:
```diff
@@ -0,0 +1,45 @@
+    Long coffeeChatInfoId,
+    Long applierUserId,
+    String applierName,
+    String status,
```
status는 enum 타입인데 혹시 string 하셨는지 궁금합니다!

geonuu09 commented on boottalk/src/main/java/com/icandoit/boottalk/coffeeChat/entity/CoffeeChatApplication.java:
```diff
@@ -0,0 +1,75 @@
+    private CoffeeChatInfo coffeeChatInfo;
+
+    @Enumerated(EnumType.STRING)
+    @Column(columnDefinition = "VARCHAR(20)", nullable = false)
```
columnDefinition 속성을 사용하신 이유가 혹시 있을까요?
메모리 할당량을 제한하여 메모리 공간 확보 때문에 하신 것인가요!
맞으시다면 저도 참고해서 수정해야겠어요😊

jooyeon0907 replied:
해당 필드를 DB에서 ENUM 타입으로 생성하지 않고, VARCAHR 로 저장하기 위해 설정하였습니다.
해당 필드(statusType)가 20자를 넘지 않을 것 같아, 데이터베이스 스키마에서 적절한 길이를 설정하기 위해 VARCHAR(20) 을 사용하였습니다!

geonuu09 reviewed (commented):
코드 잘 확인했습니다! 궁금한 부분 코멘트 남겼습니다 확인 부탁드려요~ 
그리고 패키지 분리는 저희 프로젝트 사이즈에서는 나눌 필요까진 없다고 생각됩니다! 
데일리 스크럼때 다시 조율해보면 좋을 것 같습니다!
